python-avd/pyavd/_eos_designs/structured_config/connected_endpoints/port_channel_interfaces.py

```diff
@@ -155,24 +153,36 @@ def _get_port_channel_interface_cfg(
             "struct_cfg": get(adapter, "port_channel.structured_config"),
         }
 
+        if get(adapter, "port_channel.subinterfaces"):
+            port_channel_interface.update({"switchport": {"enabled": False}})
+        else:
+            port_channel_interface.update({"switchport": {"enabled": True}})
+
         # Only switches interfaces
-        if port_channel_type == "switched":
+        if port_channel_interface["switchport"]["enabled"]:
             port_channel_interface.update(
                 {
-                    "mode": adapter.get("mode"),
                     "l2_mtu": adapter.get("l2_mtu"),
                     "l2_mru": adapter.get("l2_mru"),
-                    "vlans": adapter.get("vlans"),
-                    "trunk_groups": self._get_adapter_trunk_groups(adapter, connected_endpoint),
-                    "native_vlan_tag": adapter.get("native_vlan_tag"),
-                    "native_vlan": adapter.get("native_vlan"),
-                    "phone": self._get_adapter_phone(adapter, connected_endpoint),
                     "spanning_tree_portfast": adapter.get("spanning_tree_portfast"),
                     "spanning_tree_bpdufilter": adapter.get("spanning_tree_bpdufilter"),
                     "spanning_tree_bpduguard": adapter.get("spanning_tree_bpduguard"),
                     "storm_control": self._get_adapter_storm_control(adapter),
                 },
             )
+            port_channel_interface["switchport"].update(
```

Comment:
We need the same logic here to interpret the `mode` and configure the relevant options. Maybe we should put it in some helper function if it works the same for both ethernet and port-channels.